.github/workflows/update-schema.yml

```diff
@@ -0,0 +1,93 @@
+name: Update schema
+
+concurrency: update-schema-${{ github.ref }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: "New okp4/contracts version (repository tag or repository ref)"
+        required: true
+      draft:
+        type: boolean
+        description: "Indicate if schemas need to be only publish in draft or need to be released and published publicly"
+        required: false
+        default: false
+
+jobs:
+  update-schema:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.OKP4_TOKEN }}
+          fetch-depth: 2
+
+      - name: Check out repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.version }}
+          repository: okp4/contracts
+          token: ${{ secrets.OKP4_TOKEN }}
+          path: tmp
+
+      - name: Cache cargo registry
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            ~/.cargo/bin
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-
+
+      - name: Setup rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: 1.69
+          default: true
+          override: true
+
+      - name: Install cargo make
+        uses: davidB/rust-cargo-make@v1
+
+      - name: Build schema
+        run: |
+          cd tmp
+          cargo make schema
+
+      - name: Import GPG key
+        uses: crazy-max/ghaction-import-gpg@v5
+        with:
+          gpg_private_key: ${{ secrets.OKP4_BOT_GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.OKP4_BOT_GPG_PASSPHRASE }}
+          git_config_global: true
+          git_user_signingkey: true
+          git_commit_gpgsign: true
+
+      - name: Copy updated schema
+        run: |
+          rm -rf schema/*
+          rsync -rmv --include='*/' --include='*/schema/raw/*.json' --exclude='*' tmp/contracts/ schema/
+          find schema -type f -name "*.json" -exec sh -c 'mv "{}" "$(dirname "{}" | xargs dirname | xargs dirname)"' \;
+          git rm --ignore-unmatch --cached tmp
+          rm -rf .git/modules/tmp
+          rm -rf tmp
+
+      - name: Search for all modified files that involve the creation of PR or commit to main
+        id: changed-files
+        uses: tj-actions/changed-files@v37.6.1
```

Comment:
Is it working as expected? I thought the purpose of the action `tj-actions/changed-files` was to identify changes between different revisions or branches. However, you're simply looking to see if your repository is in a dirty state. In that case, I believe using Git would be more appropriate. What do u think?

Reply:
Maybe we can simply omit this step and let the commit step go through even if there is no changes in the schema, it'll push nothing and shouldn't fail

Reply:
I understand, this method work well on `okp4-cosmos-proto`, but you're right, is not the initial function of this action. 
@amimart Yes, maybe it's not necessary to check since the commit will not be pushed if there is no changes. Let's try this.

Reply:
I've added `file_pattern` option on the commit step to avoid push not wanted files.